nilearn/_utils/numpy_conversions.py

```diff
@@ -124,3 +126,46 @@ def as_ndarray(arr, copy=False, dtype=None, order='K'):
         raise ValueError("Type not handled: %s" % arr.__class__)
 
     return ret
+
+
+def csv_to_array(csv_path, delimiters=' \t,;', **kwargs):
+    """Read a CSV file by trying to guess its delimiter
+
+    Parameters
+    ----------
+
+    csv_path: string
+        Path of the CSV file to load.
+
+    delimiters: string
+        String containing the possible delimiters for the CSV file.
+
+    **kwargs: keyword arguments
+        The additional keyword arguments are passed to numpy.genfromtxt when
+        loading the CSV.
+
+    Returns
+    -------
+
+    array: numpy.ndarray
+        An array containing the data loaded from the CSV file.
+    """
+    if not isinstance(csv_path, _basestring):
+        raise TypeError('CSV must be a file path. Got a CSV of type: %s' %
+                        type(csv_path))
+    # First, we try genfromtxt which works in most cases.
+    array = np.genfromtxt(csv_path, **kwargs)
+
+    if array.ndim <= 1 and np.all(np.isnan(array)):
+        # If the delimiter is not known genfromtxt generates an array full of
+        # nan. In that case, we try to guess the delimiter
+        try:
+            with open(csv_path, 'r') as csv_file:
+                dialect = csv.Sniffer().sniff(csv_file.readline(), delimiters)
+        except csv.Error as e:
+            raise TypeError(
+                'Could not read CSV file [%s]: %s' % (csv_path, e.args[0]))
```

Comment:
The csv sniffer returns a csv.Error. I thought it would be more consistent to always return a TypeError when there is an error loading a CSV file. Otherwise doing a `try ... except` around this function could be cumbersome.